base/genimport/importer_load.go

```diff
@@ -71,7 +73,13 @@ func (imp *Importer) Load(dir string, paths []string, mode ImportMode, enableMod
 		return pkgs, nil
 	}
 
-	goModReplaceDirective, pkgpaths, err := findLocalPackagesOnDisk(paths)
+	// Allow local imports by module-name rather than relative path.
+	localModuleName, localAbspath, err := getModuleFileInfo(".")
+	if err != nil {
+		return nil, err
```

Comment:
For our use it's probably fine, but I suspect if we wanted to upstream this we wouldn't want to fail loading a dependency just because the local directory doesn't have itself or an ancestor with a `go.mod` file?

Reply:
Right, good point!